integrators/ordinary_differential_equations.hpp

```diff
@@ -92,6 +93,9 @@ struct AdaptiveStepSize {
   // In both cases, the new step size is chosen so as to try and make the result
   // of the next call to |tolerance_to_error_ratio| close to |safety_factor|.
   ToleranceToErrorRatio tolerance_to_error_ratio;
+  // If |max_steps|, integration will stop after |*max_steps| even if it has not
+  // reached |t_final|.
+  std::experimental::optional<std::int64_t> max_steps;
```

Comment:
I'd prefer to make `max_steps` a plain int, initialized here to `std::numeric_limits<std::int64_t>::max()`.  That would be more efficient and readable than having to check of the optional is armed.